vpr/src/place/analytical_placement/analytical_placement_flow.cpp

```diff
@@ -51,40 +55,44 @@ void run_analytical_placement_flow() {
     // Using the placement from the SA to get the fixed block locations
     std::set<AtomBlockId> fixed_blocks;
     std::map<AtomBlockId, double> fixed_blocks_x, fixed_blocks_y;
+    const VprConstraints constraints = g_vpr_ctx.mutable_floorplanning().constraints;
```

Comment:
I do not think you need the mutable version of floorplanning here. Also you should get the constraints by reference.

const VprConstraints& constraints = ...